lib/rubocop/cop/rspec/rails/minitest_assertions.rb

```diff
@@ -186,8 +119,57 @@ def assertion
             end
           end
 
+          # :nodoc:
+          class InDeltaAssertion < BasicAssertion
+            MATCHERS = %i[
+              assert_in_delta
+              assert_not_in_delta
+              refute_in_delta
+            ].freeze
+
+            NODE_MATCHER_PATTERN = <<~PATTERN
+              (send nil? {:assert_in_delta :assert_not_in_delta :refute_in_delta} $_ $_ $!{sym str}? $_?)
```

Comment:
Message can be a proc:
```ruby

# File lib/minitest/unit.rb, line 196
    def assert test, msg = nil
      msg ||= "Failed assertion, no message given."
      self._assertions += 1
      unless test then
        msg = msg.call if Proc === msg
        raise MiniTest::Assertion, msg
      end
      true
    end
    ```

Reply:
That's not unique to this though, and none of our code depends on that - we're just copying the source code as is (you might be mentioning this because we're doing `!failure_message.empty?` for this assertion, but that is checking against an _array_ which contains the matched nodes).

We should have tests for this but since it applies for every assertion we currently have I'll do it in a dedicated PR

Reply:
I’m most concerned (see my earlier comments below), is that we assume that the message (str/sym) can be passed as the third argument, and in this case, skip matching. But this would be incorrect for Minitest.
My point is to disregard the type, and always consider the third argument to be the delta (optional), and the fourth - the message (optional, no type constraints.

I agree that we’d better investigate proc messages in a separate pr. Wondering if RSpec provides a lazily evaluated failure message and we could autocorrect those.
If it can’t - we need to skip autocorrection.

Reply:
Good news that [RSpec does seem to support proc failure messages.](https://github.com/rspec/rspec-expectations/blob/96a3e5ad8a739318378a0d226a5bff307272cbd8/lib/rspec/expectations/handler.rb#L6). I would still double check this.